libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/arrow/api/MapboxRouteArrowView.kt

```diff
@@ -33,12 +35,38 @@ import com.mapbox.navigation.utils.internal.logE
  * @param options the options used for determining the rendering appearance and/or behavior.
  */
 @UiThread
-class MapboxRouteArrowView(private val options: RouteArrowOptions) {
+class MapboxRouteArrowView(
+    private var options: RouteArrowOptions,
+    private var rebuildLayersOnFirstRender: Boolean = true
+) {
+
+    /**
+     * Responsible for rendering state data generated by the MapboxRouteArrowApi class. The
+     * state data will alter the appearance of the maneuver arrow(s) on the map.
+     *
+     * Each [Layer] added to the map by this class is a persistent layer - it will survive style changes.
+     * This means that if the data has not changed, it does not have to be manually redrawn after a style change.
+     * See [Style.addPersistentStyleLayer].
+     *
+     * @param options the options used for determining the rendering appearance and/or behavior.
+     */
+    constructor(routeArrowOptions: RouteArrowOptions) : this(routeArrowOptions, true)
 
     private companion object {
         private const val LOG_CATEGORY = "MapboxRouteArrowView"
     }
 
+    /**
+     * Calling this will reinitialize the map layers with the [RouteArrowOptions].
+     *
+     * @param style the [Map] style to initialize the layers on.
+     * @param options to use for reinitializing the arrow related layers.
+     */
+    fun update(style: Style, options: RouteArrowOptions) {
+        this.options = options
+        rebuildLayersAndSources(style, options)
+    }
```

Comment:
This isn't truly needed if we're rebuilding layers and sources on instance creation anyway, let's commit to that approach first. If it would turn out not to be enough, we can consider additional changes.

Reply:
```suggestion
    /**
     * Calling this will reinitialize the map layers with the [RouteArrowOptions].
     *
     * @param style the [Map] style to initialize the layers on.
     * @param options to use for reinitializing the arrow related layers.
     */
    fun update(style: Style, value: ArrowAddedValue, options: RouteArrowOptions) {
        this.options = options
        rebuildLayersAndSources(style, options)
    }
```
Let's change to this as per conversation.

Reply:
Going back and removing this function after further convo. Re-instantiating shouldn't be a problem.